lib/api_repo.js

```diff
@@ -873,15 +926,34 @@ ApiRepo.prototype._makeProtocFunc = function _makeProtocFunc(opts, language) {
   opts = opts || {};
   opts.env = opts.env || this.opts.env || process.env;
 
+  // computeOutDir calculates the directory in which protoc should place the output.
+  function computeOutDir(prefix, language, protoRepoDir, protoPath, done) {
+    var file = readline.createInterface({
+      input: fs.createReadStream(path.join(protoRepoDir, protoPath))
+    });
+    file.on('line', function (line) {
+      var match = line.match('^package ([A-Za-z_]+(\.[A-Za-z_0-9]+)*)');
+      if (match) {
+        var pkg = match[1].replace(/\./g, '/');
+        done(path.join(prefix, language, pkg));
+      }
+    });
+  }
+
   // callProtoc invokes protoc for the given language
-  function callProtoc(protoPath, done) {
-    if (that.languages.indexOf(language) === -1) {
+  function callProtoc(repoDir, protoPath, done) {
```

Comment:
This changes the return type of makeProtocFunc. Please update JSDoc.

Also, I guess this would change the usage of this function (at the end of buildProtos).